base/v0_5_exp/translate.go

```diff
@@ -217,6 +218,48 @@ func translateLink(from Link, options common.TranslateOptions) (to types.Link, t
 	return
 }
 
+func translateSSHAuthorizedKeysLocal(from []SSHAuthorizedKeyFile, options common.TranslateOptions) (to []types.SSHAuthorizedKey, tm translate.TranslationSet, r report.Report) {
+	tr := translate.NewTranslator("yaml", "json", options)
+	tm, r = translate.Prefixed(tr, "ssh_authorized_keys_local", &from, &to)
+	c := path.New("yaml", "ssh_authorized_keys_local")
+
+	if len(from) == 0 {
+		return // no SSH keys supplied, we're done here
+	}
+	if options.FilesDir == "" {
+		r.AddOnError(c, common.ErrNoFilesDir)
+		return
+	}
+
+	for _, file := range from {
+		// calculate file path within FilesDir and check for
+		// path traversal
+		filePath := filepath.Join(options.FilesDir, string(file))
+		if err := baseutil.EnsurePathWithinFilesDir(filePath, options.FilesDir); err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+		contents, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+
+		sshKeyStrings := strings.TrimSpace(string(contents))
+		if len(contents) == 0 || len(sshKeyStrings) == 0 {
+			r.AddOnError(c, common.ErrFileEmpty)
+		}
+		tm.AddTranslation(c, path.New("json", "ssh_authorized_keys"))
```

Comment:
You're missing the array indexes of both sides.  Since the left side may produce multiple right sides, you'll want to move this into the loop below.